### PR TITLE
Remove generated dune files on "make clean"

### PR DESCRIPTION
**Pull request by yaitskov**
Fixes https://github.com/coq/coq/issues/19918

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -211,7 +211,7 @@ ireport:
 	dune build $(DUNEOPT) @install --profile=ireport
 
 clean:
-	rm -f .dune-stamp
+	rm -f $$(awk '/BEGIN REMOVE ON make clean/{flag=1; next} /END REMOVE ON make clean/{flag=0} flag' .gitignore)
```
Comment (SkySkimmer) [inline Makefile:214]: I think I'd rather duplicate the list than parse adhoc comments in gitignore

**Review by SkySkimmer — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (yaitskov) [reply Makefile]: Fixed

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
Please squaash the commits

**Comment by SkySkimmer** [comment]
ping @yaitskov 

>Please squash the commits

**Comment by SkySkimmer** [comment]
I did the squash

@coqbot merge now

**Pull request merged**